src/FilenameParsing/FileIDHelperResolutionStrategy.php

```diff
@@ -396,9 +402,22 @@ public function findVariants($tuple, Filesystem $filesystem)
 
             $folder = $helper->lookForVariantIn($parsedFileID);
             $possibleVariants = $filesystem->listContents($folder, true);
+            $possibleVariants = array_filter($possibleVariants, function ($possibleVariant) {
```

Comment:
Could use some comments in here to explain whats happening

Reply:
Done. I added some clarifications to my tortured logic.